lib/veres-one-client.js

```diff
@@ -0,0 +1,187 @@
+/*!
+ * Copyright (c) 2018 Veres One Project. All rights reserved.
+ */
+'use strict';
+
+const axios = require('axios');
+const https = require('https');
+const WebLedgerClient = require('web-ledger-client');
+
+const DEFAULT_MODE = 'test';
+
+class VeresOneClient {
+  /**
+   * @param injector {Injector}
+   *
+   * @param hostname {string} Hostname of the ledger (points to a load balancer
+   *   or a specific node).
+   * @param ledger {WebLedgerClient}
+
+   * @param [mode='test'] {string} One of 'dev'/'test'/'live'. Determines https
+   *   agent settings (does not reject unsigned certs in 'dev' mode, for
+   *   example).
+   * @param [logger]
+   * @param [httpsAgent] {Agent}
+   */
+  constructor({injector, hostname, ledger, mode, logger, httpsAgent}) {
+    this.injector = injector;
+    this.hostname = hostname;
+    if(!hostname) {
+      throw new Error('Missing ledger hostname');
+    }
+    this.ledger = ledger ||
+      new WebLedgerClient({injector, hostname, mode, logger, httpsAgent});
```

Comment:
We need any validation on `ledger` as a `WebLedgerClient` instance ... or anything of that sort?

Reply:
We could add it, though I honestly don't think it's needed. It's only there so you can inject a ledger stub for testing, for example. Passing in a ledger object is not going to be a common enough operation to need validating here.